src/lexer.coffee

```diff
@@ -256,13 +258,7 @@ exports.Lexer = class Lexer
       when /^0\d+/.test number
         @error "octal literal '#{number}' must be prefixed with '0o'", length: lexedLength
 
-    base = switch number.charAt 1
-      when 'b' then 2
-      when 'o' then 8
-      when 'x' then 16
-      else null
-
-    numberValue = if base? then parseInt(number[2..], base) else parseFloat(number)
+    numberValue = getNumberValue number
```

Comment:
Ah, I see, you wrapped into a function some code that used to be loose. Sure, it’s better as a function, but if it’s only used in the lexer let’s keep it in `lexer.coffee` I think.

Reply:
If we’re getting the number value here, couldn’t that be passed along to `nodes.coffee`? Why would we need to run the `getNumberValue` function twice for the same token?

Reply:
@GeoffreyBooth it looks like currently `lexer.coffee` is just calculating the number value for the sake of checking if it's `Infinity` or not. So you're suggesting "caching" the calculated number value on the `NUMBER` token as additional token data to avoid recalculating later? That makes sense to me in theory, maybe treat that as a separate optimization?